pgjdbc/src/main/java/org/postgresql/core/Utils.java

```diff
@@ -127,8 +103,7 @@ private static void doAppendEscapedLiteral(Appendable sbuf, String value,
         // avoid a security hazard if the reported value of
         // standard_conforming_strings is incorrect, or an error if
         // backslash_quote is off.
-        for (int i = 0; i < value.length(); ++i) {
-          char ch = value.charAt(i);
+        for (char ch : value.toCharArray()) {
```

Comment:
I wonder if this change would result array allocation and extra cpu/memory consumption issues